### PR TITLE
Introduction to deep learning

### PR DESCRIPTION
**Pull request by JudyNduati**
#### Preliminary Checks - Formatting and Structure 
- [x] Does your article follow any of the suggested structure formats? see the example [Formats here](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)
- [x] Is your article properly formatted in Markdown? - see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)
- [x] Have you used the correct folder and file structure? - see [Contribution Guidelines](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md)
- [x] Is your article over 750 words?
- [x] Have you used our preferred heading structure? - [H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)
- [x] Does your article provide enough value and detail about your topic? Articles should be clear, accurate, and fully explained.
- [x] Can your article be understood by beginners? **Assume the audience is smart but has no prior exposure to the common terminology in your article.**
- [x] Have you included a hero image? (600X400 pixels (images can be resized in paint) - also please make sure the image is from a site like www.unsplash.com)
- [ ] In your PR - add a comment that states ["This fixes #(enter your Idea Content Form #)"](https://youtu.be/TKJ4RdhyB5Y?t=278) to link and close your Idea Content (issue) when the PR is merged.
      

#### Grammar Checks
- [x] Have you spellchecked and grammar checked your article? - Tool: [Grammarly](https://grammarly.com)
- [x] Have you checked your article for readability? - Tool: [Hemmingway](http://www.hemingwayapp.com/)
- [x] Have you added sources for quotes and images that aren't yours?

### Technical Checks
- [ ] Are your code snippets properly formatted for syntax highlighting - see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting/)
- [ ] Have you used Repl.it for your code snippets?

### Contribution Guidelines
For first-time contributors and for more details, see [Contributing Guidelines](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md)

### CODE REVIEW

**Comment by JudyNduati** [comment]
I have added an index.md file, hero.jpg, a Venn diagram, and neural networks image.
Article Description
 Deep learning is a subset of Machine Learning and Artificial intelligence. This article describes the difference between ML and DL. It explains why we need Deep Learning, its applications, and how deep learning algorithms work with neural networks.

**Comment by cyberShaw** [comment]
Great work on the article, @JudyNduati 
I've done the grammar checks, readability. Since the changes were minor I've fixed those changes in commits: 680205f and 6df26bb.
We can move on to the tech review phase!

**Comment by hectorkambow** [comment]
👋  @JudyNduati  - upon running your article through our 3rd party plagiarism software - it raised a few red flags - please review the attached PDF and revisit the sentences that are highlighted and make sense to change. Thank you
🚀 
[Introduction to deep learning #372.pdf](https://github.com/section-io/engineering-education/files/5344448/Introduction.to.deep.learning.372.pdf)

**Comment by cyberShaw** [comment]
@JudyNduati if you could fix the red flags that are raised, we can move this article to the next round of review.

**Comment by JudyNduati** [comment]
@hectorkambow I have fixed the red flags. Your suggestions helped me improve the article readability

**Comment by hectorkambow** [comment]
@JudyNduati thank you for the Deep Learning article - already learning a lot by reading it. 

Upon a review of the text - some wording was a bit difficult to get through - some parts were redundant and others - hard to understand. 

We want to make sure we are using simple terms to explain the concepts - and that they flow easily. We also want to make sure we are explaining any concepts we are introducing so the audience can follow along.

Before we publish, could you review the article again - I went ahead and tried rearranging or editing the 1st paragraph or so to give you a guide of the style and tonality. 

🚀 
Thank you

**Comment by JudyNduati** [comment]
@hectorkambow I made the changes you suggested. I believe the article readability is improved and has a flow

File: articles/introduction-to-deep-learning/index.md [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+AI is accomplished by studying how the human brain operates while trying to solve a problem. ML is a branch of AI that gives machines and computers the ability to learn data without being programmed.
+
+Deep Learning is a machine learning technique that automatically extracts the useful pieces of information or makes decisions using neural networks. Deep Learning is more than a set of tools to solve practical problems. DL was made to imitate how the human brain works while making decisions.
+
+Neural Networks design depends on the composition of the brainpower. Neural Networks are trained to learn models, identify patterns, and arrange non-identical kinds of information as the human brain does.
+
+Deep Learning algorithms strive to reason the same as humans by studying data with a particular [logical framework](https://medium.com/tebs-lab/introduction-to-deep-learning-a46e92cb0022).
+
+### Machine Learning vs. Deep Learning
+Deep Learning and machine learning are both types of AI that classify data and train models.
```
Comment (hectorkambow) [inline articles/introduction-to-deep-learning/index.md:39]: I believe this vs section needs some attention as it was a bit confusing to read through - 
For example : 
"Data presentation in the system." this sentence - wasn't sure what it was telling me as the reader - had to take a moment to realize that you meant to say - "The way data is presented in ML is different from DL.

File: articles/introduction-to-deep-learning/index.md [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+Example: you have several animal images, such as birds and pets. A person would label each image to help an ML algorithm differentiate the images. Deep Learning is a bit complex. This is because using the same dataset. Deep Learning algorithms toil to evaluate variations between categories and classify them.
+
+2. A Machine Learning algorithm can take a minute or up to a few hours to finish execution. Deep Learning algorithms can go up to weeks to halt the execution. This is due to the complexity of Neural Networks. Neural networks need a large number of weights and more parameters.
+
+3. The performance of ML algorithms is better when the dataset is small or medium. Deep Learning algorithms fail to perform well on such datasets. DL algorithms instead perform exceptionally well for bigger datasets.
+
+4.	Machine Learning algorithms can perform on inexpensive. Deep Learning is a potent tool. It performs well on a powerful machine equipped with multiple GPUs, thus [providing high performance](https://hackr.io/blog/machine-learning-vs-deep-learning).
```
Comment (hectorkambow) [inline articles/introduction-to-deep-learning/index.md:48]: "Machine Learning algorithms can perform on inexpensive." 
I believe this is an unfinished thought?

Comment (hectorkambow) [reply articles/introduction-to-deep-learning/index.md:48]: The sentence "Deep Learning is a potent tool." is redundant.

File: articles/introduction-to-deep-learning/index.md [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+
+5. Machine Learning algorithms study and understand labeled data in a dataset. Deep Learning algorithms require no human intervention. This is because layers in Neural Networks place data in different concepts. Neural Networks also enable DL algorithms to learn independently.
+
+6. Deep Learning is an essential part of Machine Learning. It provides solutions to many problems in image recognition, speech recognition, and NLP. Deep Learning performs well with images, text, sound recognition as compared to Machine Learning.
+
+### How Deep Learning Works
+The functioning of the brain influences deep Learning algorithms. Neural Networks are made up of layers and nodes, as well as the brain comprises of neurons.
```
Comment (hectorkambow) [inline articles/introduction-to-deep-learning/index.md:55]: "The functioning of the brain influences deep Learning algorithms." this sentences is redundant

File: articles/introduction-to-deep-learning/index.md [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+
+Deep Learning is a subset of machine learning(ML), DL learns features and tasks directly from data such as images, text, or sound.
+
+To take it further, machine learning is a subset of artificial intelligence(AI). AI is the capability of machines and computers to mimic human intelligence and behavior.
+
+AI is accomplished by studying how the human brain operates while trying to solve a problem. ML is a branch of AI that gives machines and computers the ability to learn data without being programmed.
+
+Deep Learning is a machine learning technique that automatically extracts the useful pieces of information or makes decisions using neural networks. Deep Learning is more than a set of tools to solve practical problems. DL was made to imitate how the human brain works while making decisions.
+
+Neural Networks design depends on the composition of the brainpower. Neural Networks are trained to learn models, identify patterns, and arrange non-identical kinds of information as the human brain does.
```
Comment (hectorkambow) [inline articles/introduction-to-deep-learning/index.md:34]: I believe a section on Neural Networks would be really helpful to follow along 🚀

**Comment by hectorkambow** [comment]
@lalith1403 👋 
Good afternoon Lalith - I'm reviewing this Deep Learning article - that could benefit from a more experienced eye - would you mind taking a look and providing the author some guidance on the tech and readability. 
I've started to make some changes on the 1st paragraphs and made some single line comments - then thought you'd be a goo candidate for the tech.

File: articles/introduction-to-deep-learning/index.md [hunk 5]
```diff
@@ -0,0 +1,123 @@
...
+### Introduction
+### What is Deep Learning?
+
+![Venn diagram](/engineering-education/introduction-to-deep-learning/ai-ml-dl.jpg)<br>
+
+[*Image source*](https://towardsdatascience.com/cousins-of-artificial-intelligence-dda4edc27b55)
+
+Deep Learning is a subset of machine learning(ML), DL learns features and tasks directly from data such as images, text, or sound.
+
+To take it further, machine learning is a subset of artificial intelligence(AI). AI is the capability of machines and computers to mimic human intelligence and behavior.
```
Comment (lalith1403) [inline articles/introduction-to-deep-learning/index.md:28]: I feel to 'take it further' is redundant here. Instead, consider adding the definition and applications of machine learning after the sentence 'machine learning is a subset of AI'.

File: articles/introduction-to-deep-learning/index.md [hunk 6]
```diff
@@ -0,0 +1,123 @@
...
+
+AI is accomplished by studying how the human brain operates while trying to solve a problem. ML is a branch of AI that gives machines and computers the ability to learn data without being programmed.
```
Comment (lalith1403) [inline articles/introduction-to-deep-learning/index.md:30]: Good point. Adding the fact that machine learning takes a statistical approach to obtain patterns from data would add additional weightage to the statement being made.

File: articles/introduction-to-deep-learning/index.md [hunk 7]
```diff
@@ -0,0 +1,123 @@
...
+
+Deep Learning is a machine learning technique that automatically extracts the useful pieces of information or makes decisions using neural networks. Deep Learning is more than a set of tools to solve practical problems. DL was made to imitate how the human brain works while making decisions.
```
Comment (lalith1403) [inline articles/introduction-to-deep-learning/index.md:32]: The fact that DL is modeled after the brain is false. The neuron in the neural network is inspired by the brain. But most of the activation functions and neural architectures used in DL don't represent the architectures present in the brain. The brain is a combination of various neural architectures working together. I would avoid talking about DL imitating how the human brain works. This is more of a PR strategy to make DL more accessible and acceptable.

File: articles/introduction-to-deep-learning/index.md [hunk 8]
```diff
@@ -0,0 +1,123 @@
...
+
+Neural Networks design depends on the composition of the brainpower. Neural Networks are trained to learn models, identify patterns, and arrange non-identical kinds of information as the human brain does.
+
+Deep Learning algorithms strive to reason the same as humans by studying data with a particular [logical framework](https://medium.com/tebs-lab/introduction-to-deep-learning-a46e92cb0022).
+
+### Machine Learning vs. Deep Learning
+Deep Learning and machine learning are both types of AI that classify data and train models.
+
+1. Data presentation in the system. Machine Learning algorithms need structured data to solve data problems. Deep Learning relies on layers of Neural Networks.
```
Comment (lalith1403) [inline articles/introduction-to-deep-learning/index.md:41]: Machine learning algorithms require manual feature extraction. Deep learning algorithms perform feature extraction automatically. Consider convolutional neural networks. The feature extraction happens in the convolution step. Multiple convolutional layers perform the feature extraction and finally, these features are sent to the fully connected neural network. 

The fact that machine learning algorithms require structured data needs to be made clear. If you are referring to clean data, all algorithms depend on the quality of the data fed into them. You may consider rewording the sentence to something along the lines of: 'Machine learning algorithms require extensive data pre-processing and manual feature extraction.'

File: articles/introduction-to-deep-learning/index.md [hunk 9]
```diff
@@ -0,0 +1,123 @@
...
+Example: you have several animal images, such as birds and pets. A person would label each image to help an ML algorithm differentiate the images. Deep Learning is a bit complex. This is because using the same dataset. Deep Learning algorithms toil to evaluate variations between categories and classify them.
```
Comment (lalith1403) [inline articles/introduction-to-deep-learning/index.md:42]: Factual error: When you talk about classification, you are talking about supervised learning, which deals with labeled datasets. Therefore, irrespective of machine learning or deep learning, you would end up labeling the images for classification tasks. The difference lies in feature extraction as mentioned in the earlier comment.

Avoid using vague terms such as deep learning algorithms toil to evaluate variations between categories. Emphasizing on what variations you are talking about will enhance the readability.

File: articles/introduction-to-deep-learning/index.md [hunk 10]
```diff
@@ -0,0 +1,123 @@
...
+
+2. A Machine Learning algorithm can take a minute or up to a few hours to finish execution. Deep Learning algorithms can go up to weeks to halt the execution. This is due to the complexity of Neural Networks. Neural networks need a large number of weights and more parameters.
```
Comment (lalith1403) [inline articles/introduction-to-deep-learning/index.md:44]: Factual error: Machine learning algorithm's performance does not increase with the size of the dataset. Therefore, ML is used when there is a lack of computational power available or a small dataset. A deep learning algorithm's performance improves with the size of the dataset. Therefore, deep learning algorithms are fed petabytes of data, which required weeks of training. ML can be trained for weeks, but nothing useful will come out of it.

File: articles/introduction-to-deep-learning/index.md [hunk 11]
```diff
@@ -0,0 +1,123 @@
...
+
+3. The performance of ML algorithms is better when the dataset is small or medium. Deep Learning algorithms fail to perform well on such datasets. DL algorithms instead perform exceptionally well for bigger datasets.
+
+4.	Machine Learning algorithms can perform on inexpensive. Deep Learning is a potent tool. It performs well on a powerful machine equipped with multiple GPUs, thus [providing high performance](https://hackr.io/blog/machine-learning-vs-deep-learning).
+
+5. Machine Learning algorithms study and understand labeled data in a dataset. Deep Learning algorithms require no human intervention. This is because layers in Neural Networks place data in different concepts. Neural Networks also enable DL algorithms to learn independently.
```
Comment (lalith1403) [inline articles/introduction-to-deep-learning/index.md:50]: Factual error: Refer to the earlier comment on supervised learning. Classification requires labeled data. Be it ML or DL, we need to feed the algorithms labeled data for classification tasks. Again, you might be referring to the feature extraction process.

File: articles/introduction-to-deep-learning/index.md [hunk 12]
```diff
@@ -0,0 +1,123 @@
...
+The functioning of the brain influences deep Learning algorithms. Neural Networks are made up of layers and nodes, as well as the brain comprises of neurons.
+
+Neural Networks consist of:
+- Input layer
+- Hidden layers
+- Output layer
+
+The input layer receives input data. Hidden layers perform mathematical computations on the information. The output layer gives the result. The problem is solved based on computation from the distributive [weights of all layers](https://www.modev.com/blog/how-deep-learning-works).
+
+A good example to use is the recognition of a house image. The first layer might focus on borderlines. Another layer focus on the object colors, and another the measurements of the image.
```
Comment (lalith1403) [inline articles/introduction-to-deep-learning/index.md:64]: You need to mention the use of CNNs here. If not, the reader might get confused with ANNs. Such varied feature extraction takes place in CNNs.  Also if possible, include an image of a neural network showing the feature extraction process.

File: articles/introduction-to-deep-learning/index.md [hunk 13]
```diff
@@ -0,0 +1,123 @@
...
+
+Deep Learning systems need a powerful software. This is because they process huge amounts of data and complex computations.
```
Comment (lalith1403) [inline articles/introduction-to-deep-learning/index.md:66]: Deep learning systems require the following:
1. Loads of data
2. Computational power to process tons of data.

It is not clear when you say DL systems need powerful software.

File: articles/introduction-to-deep-learning/index.md [hunk 14]
```diff
@@ -0,0 +1,123 @@
...
+### Applications of Deep Learning
+There is massive excitement about artificial intelligence and its subsets. Here are Deep Learning applications that will govern the world.
+
+1. Self-driven cars
+
+Companies such as Google are building driver assistance services. Also, they are teaching computers how to use digital sensors instead of human sensors. These companies train Deep Learning algorithms to use [large amounts of data](https://medium.com/breathe-publication/top-15-deep-learning-applications-that-will-rule-the-world-in-2018-and-beyond-7c6130c43b01).
+
+2. Natural Language Processing
+
+Machines are taught to understand complexities associated with languages such as semantics. To achieve this, [NLP through Deep Learning](https://www.mygreatlearning.com/blog/natural-language-processing-tutorial/) plays a significant role. NLP also catch linguistic nuances and frame appropriate responses.
```
Comment (lalith1403) [inline articles/introduction-to-deep-learning/index.md:82]: I wrote the article you have linked here! Haha! good choice :) Just informing. No need to take any action here. 👍

File: articles/introduction-to-deep-learning/index.md [hunk 15]
```diff
@@ -0,0 +1,123 @@
...
+
+3. Healthcare
+
+Deep Learning is completely revolutionizing the healthcare and the medical industries. According to [Medium](https://medium.com/breathe-publication/top-15-deep-learning-applications-that-will-rule-the-world-in-2018-and-beyond-7c6130c43b01), AI has enabled healthcare and medical industries to advance tremendously.
```
Comment (lalith1403) [inline articles/introduction-to-deep-learning/index.md]: I don't think Medium is the author of the article. Kindly change the sentence according to the author's/ organization's name. According to "author's/organization's name"

File: articles/introduction-to-deep-learning/index.md [hunk 16]
```diff
@@ -0,0 +1,123 @@
...
+6. Image Recognition
+
+Image Recognition using Deep Learning aims to recognize and learn content in images. Also, Deep Learning aims to understand their surrounding. Image Recognition is mostly used in gaming and social media.
+
+7. Entertainment
+
+Entertainment industries such as [Netflix](https://www.netflix.com/ke-en/) recommends to their viewers what they need. Deep Learning enables the entertainment industry to understand the consumer's behavior. This helps to provide an exciting experience to clients. Deep Learning is revolutionizing the filmmaking process. Cameras can learn body language and conduct voice synthesis. Deep Learning also helps to emulate someone's voice to imbibe in virtual characters.
+
+### Working of Neural Networks
+Neural Networks are algorithms modeled to function like the human brain. Neural Networks is a system of hardware and software patterned after neurons operation in the human mind. Neurons are decision-makers.
```
Comment (lalith1403) [inline articles/introduction-to-deep-learning/index.md:111]: They are not modeled to function as the brain. They are simply inspired by the brain. 

Neurons are not the decision-makers. They are computational units. The decision is taken by the loss function and the optimization process.

File: articles/introduction-to-deep-learning/index.md [hunk 17]
```diff
@@ -0,0 +1,123 @@
...
+
+![Neural Networks](/engineering-education/introduction-to-deep-learning/neural-networks.jpg)<br>
+[\*image source\*](https://medium.com/coinmonks/the-artificial-neural-networks-handbook-part-1-f9ceb0e376b4)
+
+Neural Networks comprise of the input layer, hidden layers, and the output layer.
```
Comment (lalith1403) [inline articles/introduction-to-deep-learning/index.md:116]: Redundant statement.

File: articles/introduction-to-deep-learning/index.md [hunk 18]
```diff
@@ -0,0 +1,123 @@
...
+
+The input layer is where the neural network acquires data in numeral values between 0 and 1. Any data can be represented as a number, whether it is words, sound, or image. Each input neuron represents a single feature.
```
Comment (lalith1403) [inline articles/introduction-to-deep-learning/index.md:118]: The numbers can be anything. If the data is normalized, then the numeral values will be between 0 and 1. Talk about feature extractors such as SIFT, SURF, word vectors to emphasize on the fact that data can be represented as *vectors*.

File: articles/introduction-to-deep-learning/index.md [hunk 19]
```diff
@@ -0,0 +1,123 @@
...
+
+When the neurons have data, it is redirected to other neurons in the next layer (hidden layer). There are many hidden layers. Every hidden layer numerically combines all the figures it receives.
```
Comment (lalith1403) [inline articles/introduction-to-deep-learning/index.md:120]: I would rephrase the last sentence as follows: The hidden layers learn the mapping function between the input and the output. The mapping function can be thought of as the intelligence that once learned, can be used to perform the task.

File: articles/introduction-to-deep-learning/index.md [hunk 20]
```diff
@@ -0,0 +1,123 @@
...
+
+The major goal is measuring whether the inputted data has various components. Each neuron then sends data to neurons in the output layer.
```
Comment (lalith1403) [inline articles/introduction-to-deep-learning/index.md:122]: Well theoretically speaking, the goal of each neural network is defined by the loss function. It wouldn't be right to say: 'The major goal is measuring whether the inputted data has various components'. Rephrase the sentence.

**Comment by lalith1403** [comment]
@JudyNduati It's a good article. I have identified a few factual errors and a few suggestions to deliver the reader's message better. Do give all the suggestions a thorough thought and update the article. Looking forward to reading the updated article. 👍

**Comment by JudyNduati** [comment]
@lalith1403 I have worked on the revisions. Your suggestions on improving the article content and readability helped me learn more about DL, ML, and NN. I am grateful.

**Comment by lalith1403** [comment]
@JudyNduati Just to make sure there are no errors, can you check the article for grammatical errors if any on Grammarly, before we move to the next round. I have suggested a few revisions and found a few grammatical errors. Let me know once you have updated the article.

**Comment by JudyNduati** [comment]
@lalith1403 I have corrected grammatical errors in text
